flow/connectors/sql/query_executor.go

```diff
@@ -220,6 +225,11 @@ func (g *GenericSQLQueryExecutor) processRows(rows *sqlx.Rows) (*model.QRecordBa
 		}
 
 		records = append(records, record)
+		totalRowsProcessed += 1
+		if heartbeatIntervalRecords > heartbeatInterval {
```

Comment:
can this simply be `if total % hb == 0` and remove usage of hbIntervalRecords?

Reply:
fixed